planseqlearn/psl/robosuite_mp_env.py

```diff
@@ -300,6 +157,414 @@ def __init__(
             ]
         )
         self.retry = False
+    
+    def reset_precompute_sam_poses(self):
+        self.sam_object_pose = {}
+        peg_segmented = False
+        bin_segmented = False
+        for obj_name, action in self.text_plan:
+            if "cube" in obj_name:
+                frame = self.sim.render(camera_name="frontview", width=500, height=500)
+                obj_masks, _, _, pred_phrases, _ = get_seg_mask(
+                    np.flipud(frame[:, :, ::-1]),
+                    self.dino,
+                    self.sam,
+                    text_prompts=["small red cube"],
+                    box_threshold=0.35,
+                    text_threshold=0.25,
+                    device="cuda",
+                    debug=True,
+                    output_dir="sam_outputs",
+                )
+                object_mask = obj_masks[-1].cpu().detach().numpy()[0, :, :]
+                depth_map = get_camera_depth(
+                    camera_name="frontview",
+                    camera_width=500,
+                    camera_height=500,
+                    sim=self.sim,
+                )
+                depth_map = np.expand_dims(
+                    CU.get_real_depth_map(sim=self.sim, depth_map=depth_map), -1
+                )
+                world_to_camera = CU.get_camera_transform_matrix(
+                    sim=self.sim,
+                    camera_name="frontview",
+                    camera_height=500,
+                    camera_width=500,
+                )
+                camera_to_world = np.linalg.inv(world_to_camera)
+                object_pixels = np.argwhere(object_mask)
+                object_pointcloud = CU.transform_from_pixels_to_world(
+                    pixels=object_pixels,
+                    depth_map=depth_map[..., 0],
+                    camera_to_world_transform=camera_to_world,
+                )
+                self.sam_object_pose[obj_name] = np.mean(object_pointcloud, axis=0)
+            elif "door" in obj_name:
+                frame = self.sim.render(camera_name="frontview", width=500, height=500)
+                obj_masks, _, _, pred_phrases, _ = get_seg_mask(
+                    np.flipud(frame[:, :, ::-1]),
+                    self.dino,
+                    self.sam,
+                    text_prompts=["black door"],
+                    box_threshold=0.30,
+                    text_threshold=0.25,
+                    device="cuda",
+                    debug=True,
+                    output_dir="sam_outputs",
+                )
+                object_mask = obj_masks[-1].cpu().detach().numpy()[0, :, :]
+                depth_map = get_camera_depth(
+                    camera_name="frontview",
+                    camera_width=500,
```

Comment:
same comment as before, not clear why this code is so repetitive, you can condense this into helper functions and simplify this considerably